src/Microsoft.AspNet.Mvc.Core/Formatters/InputFormatter.cs

```diff
@@ -62,12 +62,17 @@ public virtual bool CanRead(InputFormatterContext context)
 
             var contentType = context.HttpContext.Request.ContentType;
             MediaTypeHeaderValue requestContentType;
-            if (!MediaTypeHeaderValue.TryParse(contentType, out requestContentType))
+            if (!MediaTypeHeaderValue.TryParse(contentType, out requestContentType) || requestContentType == null)
```

Comment:
I don't know but hit an NRE on the call to `IsSubsetOf()` just below. Seemed better to code defensively than narrow down or fix a bug in `MediaTypeHeaderValue` code I wasn't changing.